src/Services/AllowedRecipientsSanitiser.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace Digitonic\Validation\Services;
+
+use libphonenumber\NumberParseException;
+use libphonenumber\PhoneNumber;
+use libphonenumber\PhoneNumberUtil;
+use libphonenumber\PhoneNumberFormat;
+
+class AllowedRecipientsSanitiser
+{
+    protected $libPhoneNumber;
+
+    public function __construct()
+    {
+        $this->libPhoneNumber = PhoneNumberUtil::getInstance();
+    }
+
+    /**
+     * Get formatted phone number in E.164 standard without (+).
+     * This method will return false if the number is not valid.
+     *
+     * @param string $phoneNumber
+     *
+     * @return mixed
+     */
+    public function getFormattedNumber(string $phoneNumber)
+    {
+        $phoneNumberObject = $this->getPhoneNumberObject($phoneNumber);
+
+        if ($phoneNumberObject) {
+            if (in_array(
+                $this->libPhoneNumber->getRegionCodeForNumber($phoneNumberObject),
+                config('digitonic.validation.allowed_mobile_origins')
+            )) {
+                // Format number to E.164 standard and remove '+'
+                $formattedPhone = str_replace(
+                    '+',
+                    '',
+                    $this->libPhoneNumber->format($phoneNumberObject, PhoneNumberFormat::E164)
+                );
+
+                return $formattedPhone;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param PhoneNumber $phoneNumber
+     * @return string
+     */
+    public function getRegionCodeForNumber(PhoneNumber $phoneNumber): string
+    {
+        return $this->libPhoneNumber->getRegionCodeForNumber($phoneNumber);
+    }
+
+    /**
+     * Get phoneNumberObject.
+     *
+     * @param string $phoneNumber
+     *
+     * @return mixed
+     */
+    public function getPhoneNumberObject(string $phoneNumber)
+    {
+        try {
+            if (!preg_match('/[A-Za-z]/', $phoneNumber)) {
+                $phoneNumberObject = $this->libPhoneNumber->parse($this->sanitisePhoneNumber($phoneNumber), null);
+                if ($this->libPhoneNumber->isPossibleNumber($phoneNumberObject)) {
+                    return $phoneNumberObject;
+                }
+            }
+        } catch (NumberParseException $exception) {
+            // Do nothing
```

Comment:
Should we maybe log the exception, will help to trace any potential issues in the future?

Reply:
If an exception happens, it means that it is an invalid number. No need to log it in my opinion.